pinot-core/src/main/java/org/apache/pinot/core/transport/QueryRouter.java

```diff
@@ -55,25 +56,27 @@ public QueryRouter(String brokerId, BrokerMetrics brokerMetrics) {
   }
 
   public AsyncQueryResponse submitQuery(long requestId, String rawTableName,
-      @Nullable BrokerRequest offlineBrokerRequest, @Nullable Map<String, List<String>> offlineRoutingTable,
-      @Nullable BrokerRequest realtimeBrokerRequest, @Nullable Map<String, List<String>> realtimeRoutingTable,
+      @Nullable BrokerRequest offlineBrokerRequest, @Nullable Map<ServerInstance, List<String>> offlineRoutingTable,
+      @Nullable BrokerRequest realtimeBrokerRequest, @Nullable Map<ServerInstance, List<String>> realtimeRoutingTable,
       long timeoutMs) {
     assert offlineBrokerRequest != null || realtimeBrokerRequest != null;
 
     // Build map from server to request based on the routing table
     Map<Server, InstanceRequest> requestMap = new HashMap<>();
     if (offlineBrokerRequest != null) {
       assert offlineRoutingTable != null;
-      for (Map.Entry<String, List<String>> entry : offlineRoutingTable.entrySet()) {
-        Server server = new Server(entry.getKey(), TableType.OFFLINE);
+      for (Map.Entry<ServerInstance, List<String>> entry : offlineRoutingTable.entrySet()) {
```

Comment:
You are probably planning to do that in the next step, but does it make sense to get rid of the Server object, given that ServerInstance has everything that Server has (and calls resolve by host name everytime it needs to log ?)

Reply:
yes, that will be the next step. Any idea on why we have seq in ServerInstance?

Reply:
seq predates me. You approved it :)

```
commit e29933810a701e2e8c4690465ccb8774a0324180
Author: Xiang Fu <xiafu@linkedin.com>
Date:   Sun Mar 29 00:36:11 2015 -0700

    Bug fixing for federated broker test.
    1. fix logic for RealtimeColumnDataSource set range predicate as for
    realtime dictionary, dictId is from 1 to dictionary size.
    2. in MutableDictionaryReader, getString is also used for other data
    type dictionary, so change it to toString().
    3. fixing  offline ColumnDataSourceImpl would set filteredBitMap to null
    in Range query, which actually should be an empty bitMap.
    This is the root cause of DummyBlock get called.
    4. fixing NPE in HelixExternalViewBasedTimeBoundaryService.
    5. fixing time filter attching logic in BrokerRequestHandler for no
    filter request.
    6. changing attached time filters' ids to -1 and -2 to avoid potiential
    duplicated.
    
    RB=453859
    R=xiafu
    A=kgopalak,jfim
```

Reply:
We need seqId to differentiate OFFLINE table response and REALTIME table response, which is very hard to understand.. And that is the main reason why I made a new class and use TableType instead. Also, we don't need to calculate the shortHostName when routing the queries